crates/ed25519/src/circuit.rs

```diff
@@ -477,6 +494,33 @@ mod tests {
     use num_traits::Zero;
     use pasta_curves::Fp;
 
+    #[test]
+    fn alloc_affine_point_equality() {
+        for _ in 0..100 {
+            let b = Ed25519Curve::basepoint();
+            let mut rng = rand::thread_rng();
+            let scalar = rng.gen_biguint_range(&BigUint::zero(), &Ed25519Curve::order());
+            let p = Ed25519Curve::scalar_multiplication(&b, &scalar);
+
+            let mut cs = TestConstraintSystem::<Fp>::new();
+
+            let p_alloc =
+                AllocatedAffinePoint::alloc_affine_point(&mut cs.namespace(|| "alloc point p"), &p);
+            assert!(p_alloc.is_ok());
+            let p_al = p_alloc.unwrap();
+
+            let _ =
```

Comment:
This is great for positive testing, but we may want to allow the `rng` to do some negative testing for us as well.
Something like (untested pseudocode, requires defining the pt at infinity): 

```rust
let p = random_point(&mut rng);
let p2 = random_point(&mut rng);
let p_al = AllocatedAffinePoint::alloc_affine_point(&mut cs.namespace(|| "alloc point p"), &p).unwrap();
let p_al2 = AllocatedAffinePoint::alloc_affine_point(&mut cs.namespace(|| "alloc point p+p2"), &(p + p2)).unwrap();
AllocatedAffinePoint::assert_equality(&mut cs.namespace(|| "p == p + p2"), &p_al, &p2_al);
if cs.is_satisfied() {
  assert_eq!(p2, Ed25519::infinity(), "equality invariant violation!");
}
```

(see [proptest](https://altsysrq.github.io/proptest-book/intro.html) for a fancier version of this)